autoload/youcompleteme.vim

```diff
@@ -505,6 +508,8 @@ function! s:OnInsertEnter()
     return
   endif
 
+  call s:SetOmnicompleteFunc()
```

Comment:
An alternative approach would be this: https://github.com/Valloric/YouCompleteMe/compare/Valloric:master...vheon:fix-disable-filetype-defer which is to defer the set up to InsertEnter only the first time and then fall back to the original approach.